aten/src/ATen/native/quantized/decomposed/DeQuantize.cpp

```diff
@@ -0,0 +1,15 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+Tensor decomposed_dequantize_per_tensor(const Tensor& input, double scale, int64_t zero_point, ScalarType dtype) {
+  TORCH_CHECK(input.dtype() == dtype, "Expecting input to have dtype:", dtype)
+  if (dtype == at::kByte || dtype == at::kChar) {
+    return (input.to(at::kFloat) - zero_point) * scale;
```

Comment:
is the `.to` needed here when `* scale` implicitly converts the LHS to a double?

Reply:
looks like so, the test failed when we don't have this..